buildroot/share/PlatformIO/scripts/common-dependencies.py

```diff
@@ -279,17 +285,250 @@ def load_marlin_features():
 		else:
 			cmd += ['-D' + s]
 
-	cmd += ['-D__MARLIN_DEPS__ -w -dM -E -x c++ buildroot/share/PlatformIO/scripts/common-dependencies.h']
-	cmd = ' '.join(cmd)
+	cmd += ['-D__MARLIN_DEPS__ -w -dM -E -x c++']
+	depcmd = cmd + [ filename ]
+	cmd = ' '.join(depcmd)
 	blab(cmd)
 	define_list = subprocess.check_output(cmd, shell=True).splitlines()
+	return define_list
```

Comment:
Suggest: caching this result and returning the cached version when it is set.

Reply:
I'm doing it already but for the whole process. Please have a look around line 386.
I thought it wouldn't be *that* bad if we repeated the command once when the process is run for the first time (compared to moving the cache out of 2 functions).

Reply:
You should never cache compiler output, is it will change from build to build.

But you can cache configuration options names, to avoid parsing the configuration all the time. Take a look in my original PR.

Reply:
Nevermind, it wasn't that hard in fact.

Reply:
Victor: I did take a look ;-) That's why it wasn't that hard.

Reply:
> You should never cache…

I was referring to within the run of the script, don't compile twice but just once and cache the first time. Looks like the script development is coming along well….